src/loops/surrounding.lean

```diff
@@ -17,15 +20,115 @@ local notation `I` := Icc (0 : ℝ) 1
 def loop.surrounds (γ : loop F) (x : F) : Prop := 
   ∃ t w : fin (d + 1) → ℝ, surrounding_pts x (γ ∘ t) w
 
+lemma loop.surrounds_iff_range_subset_range (γ : loop F) (x : F) : 
+  γ.surrounds x ↔ ∃ (p : fin (d + 1) → F) (w : fin (d + 1) → ℝ), 
+  surrounding_pts x p w ∧ range p ⊆ range γ :=
+begin
 
+  split,
+  { exact λ ⟨t, w, h⟩, ⟨(γ ∘ t), w, h, range_comp_subset_range _ _⟩ },
+  { rintros ⟨p, w, h₀, h₁⟩,
+    rw range_subset_iff at h₁,
+    choose t ht using h₁,
+    have hpt : γ ∘ t = p := funext ht,
+    exact ⟨t, w, hpt.symm ▸ h₀⟩ }
+end
+
+set_option profiler true
 lemma surrounding_loop_of_convex_hull {f b : F} {O : set F} (O_op : is_open O) (O_conn : is_connected O) 
   (hsf : f ∈ convex_hull O) (hb : b ∈ O) : 
   ∃ γ : ℝ → loop F, continuous_on ↿γ (set.prod I univ) ∧ 
-                    ∀ t, γ t 0 = b ∧
-                    ∀ s, γ 0 s = b ∧
-                    ∀ (t ∈ I) s, γ t s ∈ O ∧
+                    (∀ t, γ t 0 = b) ∧
+                    (∀ s, γ 0 s = b) ∧
+                    (∀ (t ∈ I) s, γ t s ∈ O) ∧
```

Comment:
I wonder how many more years I'll make this mistake in statements again and again. Fortunately it usually comes to light pretty quickly when trying to prove things.